File: Ex2.py
# Функция для преобразования римского числа в натуральное число
def roman_to_int(roman_numeral):
    # Создаем словарь для соответствия римских цифр и их значений
    roman_numerals = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000}

    result = 0
    prev = 0
    for char in reversed(roman_numeral):  # Обходим римское число в обратном порядке
        current = roman_numerals[char]
        if current < prev:
            result -= current
        else:
            result += current
        prev = current

    return result

def main():
    # Получаем ввод от пользователя
    roman_numeral = input("Введите римское число: ").upper()

    # Преобразуем римское число в натуральное и выводим результат
    result = roman_to_int(roman_numeral)
    print("Результат:", result)

if __name__ == "__main__":
    main()

File: Ex2.swift
import Foundation

// Функция для преобразования римского числа в натуральное число
func romanToInt(_ romanNumeral: String) -> Int {
    // Создаем словарь для соответствия римских цифр и их значений
    let romanNumerals: [Character: Int] = [
        "I": 1,
        "V": 5,
        "X": 10,
        "L": 50,
        "C": 100,
        "D": 500,
        "M": 1000
    ]
    
    var result = 0
    var prev = 0
    for char in romanNumeral.reversed() {
        guard let current = romanNumerals[char] else { continue }
        if current < prev {
            result -= current
        } else {
            result += current
        }
        prev = current
    }
    
    return result
}

// Получаем ввод от пользователя и выводим результат
print("Введите римское число:", terminator: " ")
if let input = readLine(), !input.isEmpty {
    let result = romanToInt(input)
    print("Результат:", result)
} else {
    print("Ошибка: Введите корректное римское число.")
}
